### PR TITLE
Clean up the release process

### PR DESCRIPTION
**Pull request by marcinwyszynski**
## Description of the change

No longer push to S3. No need for the `future` branch any more. Rename flows to reflect their effect.

## Type of change

- [x] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development
- [x] Examples for new resources and data sources have been added
- [x] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [x] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [x] Reviewers have been assigned
- [x] Changes have been reviewed by at least one other engineer
- [ ] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -24,59 +28,33 @@ jobs:
...
-          SPACELIFT_API_KEY_SECRET: ${{ secrets.SPACELIFT_PROD_API_KEY_SECRET }}
-
-      - name: Set up release notes
-        run: |
-          echo ${{ steps.import_gpg.outputs.fingerprint }} > key.fingerprint
-          gpg --armor --export ${{ steps.import_gpg.outputs.keyid }} > key.asc
         
-      - name: Run GoReleaser
+      - name: Run GoReleaser to release to GitHub/HashiCorp registry
         uses: goreleaser/goreleaser-action@v2
```
Comment (peterdeme) [inline .github/workflows/release.yml:33]: might as well bump this to `v4` as well?
```suggestion
        uses: goreleaser/goreleaser-action@v4
```

Comment (marcinwyszynski) [reply .github/workflows/release.yml:33]: In principle, good idea, but can we do it in a separate PR, just in case?

Comment (peterdeme) [reply .github/workflows/release.yml:33]: sure

**Review by peterdeme — APPROVED** [review]

**Review by tomasmik — APPROVED** [review]

**Review by TheMacies — APPROVED** [review]

**Review by adamconnelly — APPROVED** [review]
:confetti_ball:

**Pull request merged**